cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -731,6 +733,11 @@ getPkgConfigDb verbosity progdb = do
     liftIO $ readPkgConfigDb verbosity progdb
 
 
+getDirectoryContentsMonitored :: FilePath -> Rebuild [FilePath]
+getDirectoryContentsMonitored dir = do
+    monitorFiles [monitorDirectory dir]
```

Comment:
And this is just an example of how the API is error prone; better to do something Shake-like where you have no access to the manual monitorFiles call ;)